app/assets/javascripts/render_ideas.js

```diff
@@ -0,0 +1,31 @@
+var getIdeas = function (requestService) {
+  return requestService.get('/api/v1/ideas');
+}
+
+var renderIdea = function (idea) {
+  return $('<div class="idea" data-id=' + idea.id + '><h3 class="idea-title" contentEditable="false">' + idea.title + '</h3>'
+  + '<h4 class="quality">' + ideaQualityCollection[idea.quality] + '</h4>'
+  + '<button class="thumbsUp">Thumbs Up</button><button class="thumbsDown">Thumbs Down</button>'
+  + '<p class="body">' + truncate(idea.body) + '</p>'
+  + '<p class="full-body" contentEditable="true" style="display: none">' + idea.body + '</p>'
+  + '<button class="delete">Delete</button></div>')
+}
+
+var renderIdeas = function (ideas) {
+  return ideas.map(renderIdea);
+}
+
+var addElementToPage = function (element) {
+  $('#ideas').prepend(element);
+}
+
+var truncate = function(string){
+  var shortendString = $.trim(string).substring(0, 99)
+  var idx = string.lastIndexOf(' ')
+
+  if (idx == -1 || string[100] === undefined) {
```

Comment:
Changed to `if (idx === -1 || string.length < 100) {// ...}` I kept the initial idx clause in case there are no spaces in a 100+ character body, it'll still display something. Edge case for sure but I live for my end users!